test/org/sagebionetworks/bridge/services/AuthenticationServiceImplTest.java

```diff
@@ -201,6 +229,58 @@ public void testSignOutWhenSignedOut() {
         assertNull(cacheProvider.getUserSessionByUserId(userId));
     }
     
+    @Test
+    public void signUpWillCreateDataGroups() {
+        authService = spy(authService);
+        optionsService = spy(optionsService);
+        authService.setOptionsService(optionsService);
+        
+        Study study = studyService.getStudy(TEST_STUDY_IDENTIFIER);
+        
+        Set<String> list = Sets.newHashSet("group1");
+        String name = TestUtils.randomName(AuthenticationServiceImplTest.class);
+        String email = "bridge-testing+"+name+"@sagebase.org";
+        
+        try {
+            SignUp signUp = new SignUp(name, email, "P@ssword1", null, list);
+
+            authService.signUp(study, signUp, true);
+            Account account = accountDao.getAccount(study, email);
+            HealthId healthId = healthCodeService.getMapping(account.getHealthId());
+            
+            verify(authService).signUp(study, signUp, true);
+            verify(optionsService).setStringSet(study, healthId.getCode(), DATA_GROUPS, list);
```

Comment:
Since optionsService is a real optionsService, I feel like this test would be more complete to verify that optionsService.getStringSet() returns the data groups we expect. This way, we're testing the behavior and not the implementation.